src/learner.cc

```diff
@@ -406,8 +406,18 @@ class LearnerConfiguration : public Learner {
   }
 
   void LoadConfig(Json const& in) override {
+    // If configuration is loaded, ensure that the model came from the same version
     CHECK(IsA<Object>(in));
-    Version::Load(in);
+    auto origin_version = Version::Load(in);
+
+    if (!Version::Same(origin_version)) {
```

Comment:
Could you please take a look into the `ModelMsg()` function? I would like to keep the difference between "model" and RDS serialization.

Reply:
I'm okay with just using ModelMsg() in the warnings.

Reply:
Replaced with `ModelMsg()`